### PR TITLE
add doctest for criterion.py

### PR DESCRIPTION
**Pull request by plrectco**
## What changes were proposed in this pull request?

(Please fill in changes proposed in this patch)

## How was this patch tested?
doctest

### CODE REVIEW

File: dl/src/main/python/nn/criterion.py [hunk 1]
```diff
@@ -251,7 +241,6 @@ def __init__(self,
 
 
 class SoftmaxWithCriterion(Criterion):
-
```
Comment (zhichao-li) [inline dl/src/main/python/nn/criterion.py]: why delete blank line here?

File: dl/src/main/python/nn/criterion.py [hunk 2]
```diff
@@ -47,21 +47,23 @@ class ClassNLLCriterion(Criterion):
     >>> classNLLCriterion = ClassNLLCriterion()
     creating: createClassNLLCriterion
     '''
-    def __init__(self, bigdl_type="float"):
-        JavaValue.__init__(self, None, bigdl_type)
+
+    def __init__(self,
```
Comment (zhichao-li) [inline dl/src/main/python/nn/criterion.py]: add another parameters:　sizeAverage: Boolean = true

File: dl/src/main/python/nn/criterion.py (hunk 1 above)
Comment (plrectco) [inline dl/src/main/python/nn/criterion.py]: for consistency purpose

**Comment by zhichao-li** [comment]
Squash and then merge it after pass the jenkins

**Pull request merged**